packages/@lwc/synthetic-shadow/src/polyfills/mutation-observer/polyfill.ts

```diff
@@ -197,8 +199,20 @@ function PatchedMutationObserver(
     return observer;
 }
 
+const sharedNodes = [document, document.head, document.body];
```

Comment:
these don't always exists, we had some issues with head and body in the past, you have to be defensive I believe.

Reply:
also, imagine the case where this is evaluated in the head, while the body hasn't been rendered/parsed yet.

Reply:
This manual list seems inherently error-prone to me... what if an app has another node that is never removed from the DOM? E.g. `<div class="container"></div>` or similar?